File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
class Solution {
    private boolean isVowel(char c) {
        return c == 'a' || c == 'A' || c == 'e' || c == 'E' || c == 'i' || c == 'I' ||c == 'o'|| c == 'O' || c == 'u' || c == 'U';
    }

    public String reverseVowels(String s) {
        char s1[] = s.toCharArray();
        int i = 0;
        int j = s.length() - 1;
        while (i < j) {
            if (!isVowel(s1[i])) {
                i++;
            } else if (!isVowel(s1[j])) {
                j--;
            } else {
                // swapping
                char t = s1[i];
                s1[i] = s1[j];
                s1[j] = t;
                i++;
                j--;
            }
        }
        return String.copyValueOf(s1);
    }
}

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.py
class Solution(object):
    def reverseVowels(self, s):
        vow = "aeiouAEIOU"
        s_list = list(s)
        i = 0
        j = len(s) - 1
    
        while i < j:
            if s_list[i] not in vow:
                i += 1
            elif s_list[j] not in vow:
                j -= 1
            else:
                s_list[i], s_list[j] = s_list[j], s_list[i]
                i += 1
                j -= 1
    
        return ''.join(s_list)
        
        